### PR TITLE
Add migration handler and various code style fixes

### PR DESCRIPTION
**Pull request by ZiyaoWei**
Master Issue: #1475 

### Motivation

The migration service needs a set of API for the clients to talk to.

Also, there are a lot of code style warnings for these files in IntelliJ which makes it harder to find real issues.

### Modifications

This adds the HTTP request handling boilerplate for the new live migration service. Note this is largely based on the Schema Registry HTTP handling code and listens to a separate port from the main Kafka listener, and in the future we should probably refactor the Schema Registry code to using this common boilerplate. Also contains some IntelliJ auto code style fixes.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

Check the box below.

Need to update docs? 
  
- [x] `no-need-doc` 
  
  Doesn't need docs at this moment, will add doc after the live migration service is done.

### CODE REVIEW

**Comment by BewareMyPower** [comment]
It's better to run `mvn checkstyle:check` in your local env to ensure the code style can pass the checkstyle check.

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java [hunk 1]
```diff
@@ -458,12 +458,10 @@ protected ApiVersionsResponse overloadDefaultApiVersionsResponse(boolean unsuppo
...
-                        case LIST_OFFSETS:
-                            // V0 is needed for librdkafka
-                            versionList.add(new ApiVersion((short) 2, (short) 0, apiKey.latestVersion()));
-                            break;
-                        default:
-                            versionList.add(new ApiVersion(apiKey));
+                        case LIST_OFFSETS ->
+                                // V0 is needed for librdkafka
+                                versionList.add(new ApiVersion((short) 2, (short) 0, apiKey.latestVersion()));
+                        default -> versionList.add(new ApiVersion(apiKey));
```
Comment (BewareMyPower) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java:464]: We'd better still keep the code compatible with Java 8 for existing code so that we can cherry-pick the PR into older branches.

BTW, it looks like the JavaDocs plugin still uses an older source as well.

```
Error:  /home/runner/work/kop/kop/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java:461: error: switch rules are not supported in -source 8
Error:                          case LIST_OFFSETS ->
Error:                                            ^
Error:    (use -source 14 or higher to enable switch rules)
```

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java [hunk 2]
```diff
@@ -1096,7 +1096,7 @@ private CompletableFuture<Pair<Errors, Long>> fetchOffset(String topicName, long
         CompletableFuture<Pair<Errors, Long>> partitionData = new CompletableFuture<>();
 
         topicManager.getTopic(topicName).thenAccept((perTopicOpt) -> {
-            if (!perTopicOpt.isPresent()) {
+            if (perTopicOpt.isEmpty()) {
```
Comment (Demogorgon314) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java:1099]: The `Optional#isEmpty` is introduced in JDK11. Do we need cherry-pick this PR to 2.9.3 or 2.10.1? Since they are using JDK1.8 to run the tests.

File: tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandlerTest.java [hunk 3]
```diff
@@ -1165,7 +1155,7 @@ public void testBrokerHandleTopicMetadataRequestAllowAutoTopicCreation(boolean b
             String topicName = "kopBrokerHandleTopicMetadataRequest-" + brokerAllowAutoTopicCreation + "-"
                     + overrideNameSpaceAutoTopicCreation + "-"
                     + allowAutoTopicCreationInRequest;
-            List<String> kafkaTopics = Arrays.asList(topicName);
+            List<String> kafkaTopics = List.of(topicName);
```
Comment (BewareMyPower) [inline tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandlerTest.java:1158]: Please revert this change because this PR could be cherry-picked to older branches that are compatible with Java 8

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandlerTest.java (hunk 3 above)
Comment (BewareMyPower) [reply tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandlerTest.java:1158]: Just found we can use `Collections.singletonList` here because the list can be immutable.

Comment (Demogorgon314) [reply tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandlerTest.java:1158]: +1

**Comment by BewareMyPower** [comment]
@ZiyaoWei Just found https://github.com/streamnative/kop/pull/1476 is merged into this branch, did you intend to merge to master?

**Comment by BewareMyPower** [comment]
IMO, it's not a good way for development. The CI won't run if the PR is not merged into master. 

image

So after #1476 is merged, this PR cannot pass the code style check.

It's better to change the target branch of all your PRs to master, including the draft PR https://github.com/streamnative/kop/pull/1513. If one PR (B) relies on another PR (A), you can include the changes in B first, then after A is merged, you can adjust code in branch B and push the extra changes to B by force.

This could also help verify the PR dependency so reviewers can know which PR should be reviewed first.

**Comment by BewareMyPower** [comment]
As you can see, if some other PRs (I see https://github.com/streamnative/kop/pull/1499 as well) are merged into this branch, this PR could become very huge and it's hard to see new changes comparing with the other PRs, which leads to the difficulty to review this PR.

**Review by BewareMyPower — APPROVED** [review]
